src/main/java/com/patchworkmc/manifest/converter/accesstransformer/AccessTransformerConverter.java

```diff
@@ -0,0 +1,98 @@
+package com.patchworkmc.manifest.converter.accesstransformer;
+
+import java.util.Map;
+
+import net.patchworkmc.manifest.accesstransformer.v2.ForgeAccessTransformer;
+import net.patchworkmc.manifest.accesstransformer.v2.Transformed;
+import net.patchworkmc.manifest.accesstransformer.v2.TransformedClass;
+import net.patchworkmc.manifest.accesstransformer.v2.TransformedField;
+import net.patchworkmc.manifest.accesstransformer.v2.TransformedMethod;
+import net.patchworkmc.manifest.accesstransformer.v2.TransformedWildcardMember;
+import net.patchworkmc.manifest.accesstransformer.v2.flags.AccessLevel;
+import net.patchworkmc.manifest.accesstransformer.v2.flags.Finalization;
+
+import com.patchworkmc.mapping.IntermediaryHolder;
+
+/**
+ * Takes a {@link ForgeAccessTransformer} and spits out an AccessWidener/v1 file.
+ */
+public class AccessTransformerConverter {
+	private AccessTransformerConverter() {
+		// NO-OP
+	}
+
+	public static byte[] convertToWidener(ForgeAccessTransformer accessTransformer, IntermediaryHolder holder) {
+		StringBuilder sb = new StringBuilder();
+		sb.append("accessWidener\tv1\t").append("intermediary").append('\n');
+		sb.append("#\tAutomatically generated by Patchwork Patcher.\n");
+		sb.append("#\tThe original (but remapped) AT entry is roughly preserved here for debugging.\n");
+		sb.append("#\tThey may be stripped in future releases.\n\n");
+
+		for (TransformedClass targetClass : accessTransformer.getClasses()) {
+			if (targetClass.getAccessLevel() != AccessLevel.KEEP || targetClass.getFinalization() != Finalization.KEEP) {
+				writeClass(sb, targetClass);
+			}
+
+			writeWildcards(sb, targetClass.getName(), targetClass.getFieldWildcard(),
+						targetClass.getMethodWildcard(), holder);
+
+			targetClass.getFields().forEach(field -> writeField(sb, field, holder));
+			targetClass.getMethods().forEach(method -> writeMethod(sb, method));
+		}
+
+		return sb.toString().getBytes();
+	}
+
+	private static void writeClass(StringBuilder sb, TransformedClass targetClass) {
+		// Modifier word
+		sb.append(targetClass.getFinalization() == Finalization.REMOVE ? "extendable" : "accessible");
+		// Target
+		sb.append("\tclass\t").append(targetClass.getName()).append('\n');
+	}
+
+	private static void writeWildcards(StringBuilder sb, String owner, TransformedWildcardMember fields, TransformedWildcardMember methods, IntermediaryHolder holder) {
+		for (Map.Entry<String, IntermediaryHolder.Member> entry : holder.getMappings(owner).entrySet()) {
+			IntermediaryHolder.Member member = entry.getValue();
+
+			if (member.isField && fields != null) {
+				// We don't write the debug because it's already been done
+				writeField(sb, owner, member.name, member.descriptor, fields);
+			} else if (methods != null) {
+				writeMethod(sb, owner, member.name, member.descriptor, methods);
+			}
+		}
```

Comment:
No need to scan all methods if there are no method wildcards and no need to scan all fields if there are no field wildcards, that would be a good cleanup of the logic (pending the splitting of the mappings in IntermediaryHolder)